sql-plugin/src/main/scala/com/nvidia/spark/rapids/JoinGatherer.scala

```diff
@@ -307,7 +307,6 @@ class LazySpillableColumnarBatchImpl(
           spill = Some(SpillableColumnarBatch(cached.get,
             SpillPriorities.ACTIVE_ON_DECK_PRIORITY))
         } finally {
-          // Putting data in a SpillableColumnarBatch takes ownership of it.
```

Comment:
Is this no longer true?

Reply:
I was debugging something here, and I must have forgotten to undo this change. coming up

Reply:
but to answer your question, the spill framework takes ownership, always